src/domain/Utilisateur.ts

```diff
@@ -91,6 +91,12 @@ export abstract class Utilisateur extends Entity<UtilisateurState> {
     return 'utilisateurNonAutoriseAChangerSonRole'
   }
 
+  mettreAJourLaDateDeDerniereConnexion(date: Date): Result<UtilisateurFailure> {
+    return Exception.toResult<UtilisateurFailure>(() => {
+      this.#derniereConnexion = new Date(date)
+    })
+  }
```

Comment:
`Exception.toResult<Failure>` "cache" un `try/catch` sur l'exécution d'une fonction susceptible de lever une exception. Par exemple, la méthode `changerTelephone` s'en sert parce qu'elle fait un `new Telephone`, le constructeur de `Telephone` pouvant lever une exception.  
Cette méthode `mettreAJourLaDateDeDerniereConnexion` instancie une `Date`, or :
1. le constructeur de `Date` ne lève pas d'exception, [mais peut retourner une date invalide](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#return_value) ; donc le recours à `Exception.toResult` n'interceptera jamais d'exception dans ce cas, ne renverra aucune `UtilisateurFailure`, et est donc inutile,
2. il n'est pas nécessaire d'instancier une nouvelle `Date` à partir d'une `Date` sauf si ton intention est de la cloner ; `this.#derniereConnexion = date` revient au même que `this.#derniereConnexion = new Date(date)`,
3. a t-on besoin d'un invariant sur `#derniereConnexion` ? Par exemple, faut il que sa valeur soit postérieure à sa valeur précédente ? Dans ce cas, une nouvelle `UtilisateurFailure` est nécessaire. Dans le cas contraire, le recours à `Exception.toResult` est inutile.